crates/anstyle-roff/src/lib.rs

```diff
@@ -0,0 +1,229 @@
+            assert_eq!(RgbColor(0, 0, 255).as_hex().as_str(), "#0000ff");
+        }
+    }
+}
+
+mod sealed {
+    pub(crate) trait Sealed {}
+}
+
+trait Ext: sealed::Sealed {
+    fn to_roff(self) -> RoffStyle;
+}
+
+impl sealed::Sealed for anstyle::Style {}
+
+impl Ext for anstyle::Style {
+    fn to_roff(self) -> RoffStyle {
+        to_roff(self)
+    }
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct RoffStyle {
+    fg: Option<roff_styles::Color>,
+    bg: Option<roff_styles::Color>,
+    effects: anstyle::Effects,
+    text: Option<String>,
+}
+
+impl RoffStyle {
+    pub fn new() -> Self {
+        Self {
+            fg: None,
+            bg: None,
+            effects: anstyle::Effects::default(),
+            text: None,
+        }
+    }
+
+    /// Add text to be styled
+    pub fn text(&mut self, txt: String) -> &mut Self {
+        self.text = Some(txt);
+        self
+    }
+
+    fn fg(&mut self, fg: Option<roff_styles::Color>) -> &mut Self {
+        self.fg = fg;
+        self
+    }
+    fn bg(&mut self, bg: Option<roff_styles::Color>) -> &mut Self {
+        self.bg = bg;
+        self
+    }
+
+    fn effects(&mut self, effects: anstyle::Effects) -> &mut Self {
+        self.effects = effects;
+        self
+    }
+
+    /// Render the style, with associated text as Roff Document
+    pub fn render(&self) -> String {
+        let mut doc = Roff::new();
+        doc.extend([set_color((&self.fg, &self.bg)), self.set_effects()]);
+        doc.to_roff()
+    }
+
+    fn set_effects(&self) -> Roff {
+        // Roff (the crate) only supports these inline commands
+        // Bold
+        // Italic
+        // Roman (plain text)
+        // If we want more support, or even support combined formats, we will need
+        // to push improvements to roff upstream or implement a more thorough roff crate
+        // perhaps by spinning off some of this code
+        let mut doc = Roff::new();
+        if let Some(ref txt) = self.text {
+            if self.effects.contains(anstyle::Effects::BOLD) {
+                doc.text(vec![bold(txt)]);
+                return doc;
+            }
+            if self.effects.contains(anstyle::Effects::ITALIC) {
+                doc.text(vec![italic(txt)]);
+                return doc;
+            }
+            if self.effects.is_plain() {
+                doc.text(vec![roff::roman(txt)]);
+                return doc;
+            }
+        }
+        doc
+    }
+}
+
+/// Generate A RoffStyle from Style
+///
+/// ```rust
+/// use anstyle::{Color, RgbColor};
+///
+/// let style = anstyle::Style::new()
+///     .fg_color(Some(Color::Rgb(RgbColor(255, 0, 170))));
+///
+/// let roff_style = anstyle_roff::to_roff(style);
+/// let expected = r#".defcolor hex_#ff00aa rgb #ff00aa
+/// .gcolor hex_#ff00aa
+/// .fcolor default
+/// "#;
+/// assert_eq!(roff_style.render(), expected);
+/// ```
+pub fn to_roff(style: anstyle::Style) -> RoffStyle {
```

Comment:
My assumption had been that we'd be parsing a `&str` that contains ANSI escape codes and converting that to roff.

While `clap` currently deals with types that are similar to `anstyle`, the plan is to update it to always be writing out ANSI escape codes and then reinterpret them as needed.

Reply:
I was patterning this off of `anstyle-yansi` and `anstyl-owo-colors`, with the idea of separating out the stream parsing code into a PR for #4. But now, if I'm understanding correctly, you see this as different than those two crates. `anstyle-yansi`/`owo-colors` are there to provide conversion to other formats, but not necessarily create output, which is what we want from `roff`. That makes sense now that I've thought about it, but want to make sure I'm understanding correctly.

Reply:
Yes, those crates are for integration.  For example, you want `anstyle` in your public API but you are using `yansi` for formatting your strings.

What we are needing for clap is the direct conversion of ansi escape codes to roff.

And I forgot earlier, clap already has [ansi code generation](https://docs.rs/clap/latest/clap/builder/struct.StyledStr.html#method.ansi)

Reply:
I've pushed a new version of this that privately implements a stream parser and uses it. I thought it would be best to put it in the Roff Stream crate for now, while we test it out and then if you like it, I can extract it into the anstyle crate itself for use. It is fairly simple for now.